### ISSUE TITLE
Nice-to-have:  a progress bar for precompiling

### ISSUE THREAD

**Issue by krrutkow** [JuliaLang/julia#35153]
Just a thought for an improved waiting experience is to show progress rather than simply `Precompiling ...`.  Even if its not representative of time, but completed precompiled packages out of the total set would be helpful, like `1/N packages`.  Or perhaps `Time for a coffee break` if many packages need to be precompiled.  Thanks!

**Comment by eschnett** [comment]
I was about to open a feature request, but then I found this. Here is what I would have written, it has a few additional ideas:

Julia pauses noticeably when packages are loaded, precompiled, or when test cases are run. It would be nice (... it would make me more patient ...) if Julia was outputting some eye candy / progress information while pausing.

For example, when running test cases, I assume a new kernel is started, and then packages are loaded. Julia could output descriptions such as "starting kernel...", "loading package X (1/n)...", "precompiling file Y...", "compiling function f...", etc.

In my mind, such output would update about every second, overwriting the previous output, and once Julia is done, this progress information would be deleted. (This is possible with most terminals.)

**Comment by pablosanjose** [comment]
See some previous discussion of this in https://discourse.julialang.org/t/feature-request-compiling-indicator/17979

**Comment by pablosanjose** [comment]
There was some previous discussion of this, particularly for function compilation, in https://discourse.julialang.org/t/feature-request-compiling-indicator/17979

**Comment by oxinabox** [comment]
Was this closed by https://github.com/JuliaLang/Pkg.jl/pull/2137 ?

**Comment by pablosanjose** [comment]
Good fishing! I'd say the OP's issue is now addressed, yes. Not the function compilation thing, of course, but that was argued to be potentially too intrusive. I'd personally close this issue.
